### ISSUE TITLE
Updating NC 14.0.3 fails with other existing custom directories

### ISSUE THREAD

**Issue by TRILOS** [nextcloud/updater#201]
Updating NC 14.0.3 to .4 failed at "Check for expected files":
"The following extra files have been found:
o [custom pathname for file repository]
o .well-known"

Update does not continue.

1. These are not "files" but directories. Please be exact.

2. Custom pathname for file repository has been defined, Update has to respect this.

3. ".well know" is a required directory for automatic verification to get a Let´s Encrypt SSL certificate by a widespread Plesk Panel extension. Update should tolerate this.

4. These are two conditions which should not break the update process. Using a software to keep the personal data in own hands should per se not deny the user´s wish.

5. Update routine of NC 14.0.03 seems to be unable to continue. May be a patch file is required.

6. This issue seems to be very, very old, see https://github.com/nextcloud/updater/issues/92

**Comment by danielkesselberg** [comment]
> .well know" is a required directory for automatic verification to get a Let´s Encrypt SSL certificate by a widespread Plesk Panel extension. Update should tolerate this.

https://github.com/nextcloud/updater/pull/183 (not sure if this feature is available for nextcloud 14)

> Custom pathname for file repository has been defined, Update has to respect this.

- directory is not inside the nextcloud root :heavy_check_mark: 
- directory name is `data` :heavy_check_mark: 
- directory inside nextcloud root but not called `data` :x: 

https://github.com/nextcloud/updater/blob/e8aac7c5b57653ef28540e01e99a604d882501e4/index.php#L293

Name of the directory should be added to this list. Not sure howto solve this properly because `datadirectory` could be an absolute or relative value and inside or outside the nextcloud root. I would always recommend to move the directory outside the webroot but i agree with you that the updater should not care about it :thinking:

**Comment by iromeister** [comment]
At least the updater should give the option to ignore existing files and directories. A warning would be OK for me.

**Comment by mfeske** [comment]
I don't get a query whether I want to ignore the files. A change to the updater / index.php only affects the web update, what do I have to do so that files at /updater/updater.phar -n are ignored

**Comment by AlphaDE** [comment]
I support @iromeister comments, the web updater should warn and allow to completely ignore those directories.

**Comment by seraphyn** [comment]
You could put these directories into the folder data and set an alias on your web server to be delivered correctly

Merry Christmas

**Comment by AlphaDE** [comment]
@seraphyn I disagree... the /stats directory is created by ISPConfig in the web-root, it's not a question of an alias in the webserver config. Why is it so difficult to let the user ignore such a warning (and consequently exclude these directories from any further processing) instead of forcing him to move directories back and forth?

**Comment by kesselb** [comment]
> Why is it so difficult to let the user ignore such a warning (and consequently exclude these directories from any further processing) instead of forcing him to move directories back and forth?

A way to skip the unexpected file warning is not implemented. 
The number of reports for this problem is low and therefore not on our roadmap.
